OpenLIFUTransducerTracker/OpenLIFUTransducerTracker.py

```diff
@@ -204,6 +212,182 @@ def updateInputOptions(self):
+            markupsWidget.setMRMLScene(slicer.mrmlScene)
+            markupsNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLMarkupsFiducialNode")
+            markupsNode.GetDisplayNode().SetViewNodeIDs([self.photoscanViewWidget.mrmlViewNode().GetID()])
+            markupsWidget.setCurrentNode(slicer.mrmlScene.GetNodeByID(markupsNode.GetID()))
+    
+        def onDialogFinished():
+            # Turn off model visibility
+            photoscan.toggle_model_display(False)
+
+        # Connect buttons and signals
+        self.photoscanPreviewDialogUI.photoscanApprovalButton.clicked.connect(onPhotoscanApproveClicked)
+        self.photoscanPreviewDialogUI.placeLandmarksButton.clicked.connect(onPlaceLandmarksClicked) 
+        self.photoscanPreviewDialog.finished.connect(onDialogFinished)
+
+        # Display dialog
+        self.photoscanPreviewDialog.exec_()
+        self.photoscanPreviewDialog.deleteLater() # Needed to avoid memory leaks when slicer is exited. 
+
+    def _create_threeDview_widget(self):
+        
+        # Layout name is used to create and identify the underlying view node 
+        layoutName = "PhotoscanPreview"
+        layoutLabel = "Photoscan Preview"
+        layoutColor = [0.97, 0.54, 0.12] # Orange
+        # ownerNode manages this view instead of the layout manager (it can be any node in the scene)
+        viewOwnerNode = slicer.mrmlScene.AddNewNodeByClass("vtkMRMLScriptedModuleNode")
+
+        # Create a view node if it hasn't been previously created
+        viewNode = slicer.util.getFirstNodeByClassByName('vtkMRMLViewNode','ViewPhotoscan')
+        if not viewNode:
+            viewLogic = slicer.vtkMRMLViewLogic()
+            viewLogic.SetMRMLScene(slicer.mrmlScene)
+            viewNode = viewLogic.AddViewNode(layoutName)
+            viewNode.SetName('ViewPhotoscan')
+            viewNode.SetLayoutLabel(layoutLabel)
+            viewNode.SetLayoutColor(layoutColor)
+            viewNode.SetAndObserveParentLayoutNodeID(viewOwnerNode.GetID())
+
+            # Customize view node. 
+            viewNode.SetBackgroundColor(0.98, 0.9,0.77) # shades of orange
+            viewNode.SetBackgroundColor2(0.98,0.58,0.4)
+            viewNode.SetBoxVisible(False) # Turn off bounding box visibility
+            viewNode.SetAxisLabelsVisible(False) # Turn off axis labels visibility
+
+        # Create widget
+        viewWidget = slicer.qMRMLThreeDWidget()
+        viewWidget.setMRMLScene(slicer.mrmlScene)
+        viewWidget.setMRMLViewNode(viewNode)
+    
+        return viewWidget
+
+    def _display_photoscan_in_widget(self, photoscan: SlicerOpenLIFUPhotoscan, viewWidget: qMRMLThreeDWidget) -> None:
+
+        photoscan_view_node = viewWidget.mrmlViewNode()
+
+        # IDs of all the view nodes in the main Window. This excludes the photoscan widget's view node
+        views_mainwindow = [node.GetID() for node in slicer.util.getNodesByClass('vtkMRMLViewNode') if node.GetName() != photoscan_view_node.GetName()]
```

Comment:
```suggestion
        views_mainwindow = [node.GetID() for node in slicer.util.getNodesByClass('vtkMRMLViewNode') if node.GetID() != photoscan_view_node.GetID()]
```

May be a rare occurrence, but aren't node IDs guaranteed to be unique, whereas names _can_ (usually by negligence) be duplicated?

Reply:
good point!